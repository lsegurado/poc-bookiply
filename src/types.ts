
export type ChannelType = 'AIRBNB' | 'HOLIDU' | 'BOOKINGCOM';

/**The review returned from the store */
export type ReviewType = {
    /**Id autogenerated in the store */
    id: string,
    /**Review header */
    headline: string,
    /**Review comment */
    comment: string,
    /**Review author */
    author: string,
    /**Review positive feedback */
    positiveFeedback: string | null,
    /**Review negative feedback */
    negativeFeedback: string | null,
    /**Review score */
    score: number,
    /**Review channel in uppercase */
    channel: ChannelType,
    /**Review date - example: 2020-08-11T12:20:02.340Z */
    publishedAt: string
}

export type GetReviewsParamsType = Partial<{
    /**An array of channels in uppercase */
    channel: ChannelType[],
    score: number[],
    _page: number,
    _limit: number
}>

/**The review returned from the back-end */
export type GetReviewsResponseType = Omit<ReviewType, 'id'>

/**Parameters that can be sent to the api */
export type ApiParamsType = Record<string, undefined | number | string | (number | string)[]>