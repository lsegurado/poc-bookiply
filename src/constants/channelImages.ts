import AIRBNB from '../assets/AIRBNB.svg';
import BOOKINGCOM from '../assets/BOOKINGCOM.svg';
import HOLIDU from '../assets/HOLIDU.svg';
import { ChannelType } from '../types';

export const channelImages: Record<ChannelType, string> = {
    AIRBNB,
    BOOKINGCOM,
    HOLIDU
};
