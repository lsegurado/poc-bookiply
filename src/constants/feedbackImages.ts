import thumbUp from '../assets/thumb-up.svg';
import thumbDown from '../assets/thumb-down.svg';

/**Images related to positive / negative feedback */
export const feedbackImages = {
    thumbUp,
    thumbDown
}