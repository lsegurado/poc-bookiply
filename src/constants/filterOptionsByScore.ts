/**Predefined options for filtering by score */
export const filterOptionsByScore = {
    /**Minimum score to look for */
    availableScoringFilters: [4.5, 4.0, 3.5, 3.0],
    /**Maximum score */
    maximumScore: 5,
    /**Precision used to send the filter by scores to the API */
    filterAccuracy: 0.1,
    /**Accuracy to be displayed by the stars in the filter */
    starsPrecision: 0.5
};
