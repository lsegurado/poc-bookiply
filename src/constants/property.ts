import background from '../assets/background.png';

export const property = {
    name: 'La Casa de las Flores',
    id: '091021',
    background
}