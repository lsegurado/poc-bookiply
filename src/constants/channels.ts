import { ChannelType } from "../types";
import { channelImages } from "./channelImages";

/**A channel list */
export const channels = Object.keys(channelImages) as ChannelType[]