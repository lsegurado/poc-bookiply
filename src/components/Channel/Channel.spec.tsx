import { render, RenderResult } from "@testing-library/react";
import { Channel } from ".";
import { channelImages } from "../../constants/channelImages";
import { ChannelType } from "../../types";

describe('Channel component', () => {
    Object.entries(channelImages).forEach(([channel, imageURL]) => {
        describe(`${channel} channel image`, () => {
            let el: RenderResult;
            beforeEach(() => {
                el = render(<Channel channelName={channel as ChannelType} />);
            })
            it('should match snapshot', () => {
                expect(el).toMatchSnapshot();
            })
            it(`should render ${channel} channel image`, () => {
                expect(el.getByRole('img').getAttribute('src')).toEqual(imageURL);
            })
        })
    });
});