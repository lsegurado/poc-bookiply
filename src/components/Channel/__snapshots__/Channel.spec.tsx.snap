// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel component AIRBNB channel image should match snapshot 1`] = `
<div>
  <img
    alt="AIRBNB"
    src="AIRBNB.svg"
  />
</div>
`;

exports[`Channel component BOOKINGCOM channel image should match snapshot 1`] = `
<div>
  <img
    alt="BOOKINGCOM"
    src="BOOKINGCOM.svg"
  />
</div>
`;

exports[`Channel component HOLIDU channel image should match snapshot 1`] = `
<div>
  <img
    alt="HOLIDU"
    src="HOLIDU.svg"
  />
</div>
`;
